app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -59,14 +75,40 @@ def create_customer
         email: user.email,
         metadata: { '3scale_account_reference' => buyer_reference }
       }
-
-      Stripe::Customer.create(customer_params, api_key).tap { |stripe_customer| payment_detail.update(credit_card_auth_code: stripe_customer.id) }
+      Stripe::Customer.create(customer_params, api_key).tap do |stripe_customer|
+        payment_detail.update(credit_card_auth_code: stripe_customer.id)
+      end
     end
 
     def api_key
       payment_gateway_options.fetch(:login)
     end
 
+    def reset_stripe_api_key
+      Stripe.api_key = nil
+    end
+
+    def latest_payment_method_id_for_customer
+      latest_payment_method = Stripe::PaymentMethod.list(
+        customer: customer.id,
+        type: 'card',
+        limit: 1
+      ).data.first&.id
+    end
```

Comment:
I don't think this is a reliable method to get the right payment mehtod :thinking: 
Are we sure about the order of the list? Maybe the first one added will be returned here, and not the one that was added last?

What we actually care about is the payment method which is identified by `payment_method_id` that we store in our database. So, I think we need to retrieve that one (I guess it would be something like `Stripe::PaymentMethod.retrieve(current_account.payment_detail.payment_method_id, api_key)`), and update it instead.